074-connector-offsets-support.md

```diff
@@ -0,0 +1,264 @@
+  ```
+  The operator will leave the annotation on the KafkaConnector resource until either the patch operation succeeds, or the user removes the annotation.
+  This means the operator will retry the patch on every reconciliation, allowing the condition to remain present for the user to see.
+* Strimzi will shortcut and automatically fail to do the reset if the KafkaConnector resource does not have it's `state` set as `stopped`.
+  Similar to if Connect returns on error, the operator will add a condition stating that the operation has failed because the connector is not stopped and therefore offsets cannot be altered, and leave the annotation on the resource.
+  The user can update the KafkaConnector to stop the connector and alter the offsets at the same time.
+  In that case the operator will first stop the connector, and once that API call returns, then it will make the call to alter the offsets.
+
+### Resetting offsets
+
+To reset the offsets for a particular connector the user will annotate the KafkaConnector resource with the new annotation `strimzi.io/connectorOffsets` set to `reset`.
+When the operator observes this annotation it will use the DELETE /connectors/{connector}/offsets endpoint to reset all offsets for the connector.
+When the reset has been completed the operator will remove this annotation from the resource.
+
+If the request to the Connect API fails the operator will add a condition to the KafkaConnector CR status with a warning message that includes the response from the API, e.g.
+  ```yaml
+  - lastTransitionTime: "2024-06-04T08:44:15.913138115Z"
+    message: Failed to alter the connector offsets due to "message from endpoint".
+    reason: ResetOffsets
+    status: "True"
+    type: Warning
+  ```
+The operator will leave the annotation on the KafkaConnector resource until either the delete operation succeeds, or the user removes the annotation.
+This means the operator will retry the delete on every operation, allowing the condition to remain present for the user to see.
+
+Strimzi will shortcut and automatically fail to do the reset if the KafkaConnector resource does not have it's `state` set as `stopped`.
+Similar to if Connect returns on error, the operator will add a condition stating that the operation has failed because the connector is not stopped and therefore offsets cannot be reset, and leave the annotation on the resource.
+The user can update the KafkaConnector to stop the connector and reset the offsets at the same time.
+In that case the operator will first stop the connector, and once that API call returns, then it will make the call to reset the offsets.
+
+## Affected/not affected projects
+
+This only affects the Connect parts of the cluster-operator.
+
+## Compatibility
+
+If a `KafkaConnectorOffsets` resource is created against a Kafka Connect version that does not support the APIs, the API calls will fail and the operator will put a suitable message in the status to indicate what has happened.
```

Comment:
I guess this is not valid anymore because the resource is not used?

Also, the KIP JIRA suggests it is supported from Kafka 3.6.0, in which case it should be supported in all versions we support. Or is there some catch?

Reply:
Yes I think Kafka 3.6.0 has all the APIs this relies on. 
Kafka 3.5.0 introduced `PUT /connectors/{connector}/pause` and `GET /connectors/{connector}/offsets`. The `PATCH /connectors/{connector}/offsets` and `DELETE /connectors/{connector}/offsets` were added in 3.6.0.